primitives/utility/src/lib.rs

```diff
@@ -297,6 +297,43 @@ impl<
 	}
 }
 
+pub struct XcmFeeManagerToAccount<FungiblesMutateAdapter, AccountId, ReceiverAccount>
+where
+	FungiblesMutateAdapter: xcm_executor::traits::TransactAsset,
+{
+	_phantom: PhantomData<(FungiblesMutateAdapter, AccountId, ReceiverAccount)>,
+}
+
+impl<FungiblesMutateAdapter, AccountId, ReceiverAccount> FeeManager
+	for XcmFeeManagerToAccount<FungiblesMutateAdapter, AccountId, ReceiverAccount>
+where
+	FungiblesMutateAdapter: xcm_executor::traits::TransactAsset,
+	AccountId: Clone + Into<[u8; 32]>,
+	ReceiverAccount: frame_support::traits::Get<Option<AccountId>>,
+{
+	fn is_waived(origin: Option<&MultiLocation>, _: FeeReason) -> bool {
+		matches!(origin, Some(MultiLocation { parents: 1, interior: Here }))
```

Comment:
This isn't exactly what I wanted. This makes it only useable for system parachain.
It will be better if you pass a generic `WavedOrigin: Contains<(Option<MultiLocation>, FeeReason)>` and have a `ParentOrigin`

Reply:
Tbh I think that it's simpler to just write a different FeeManager implementation if you want custom logic. It's only 2 functions and if we're delegating one of them, are we implementing much? I think I'd prefer it stays simpler and just used as a reference example of how to implement a FeeManager. Indirection is not without cost in terms of comprehension.

Reply:
you have a point. it is just a lot of XCM is built with this pattern.